syntheseus/search/analysis/route_extraction.py

```diff
@@ -189,3 +189,40 @@ def min_cost_routes(
             break
         else:
             yield route
+
+
+def _route_time_cost(nodes, graph) -> float:
+    """
+    Cost function for routes that is the maximum timestamp of any node in the route
+    (or inf if it is not solved).
+    """
+    if _route_has_solution(nodes, graph):
+        return max(n.creation_time.timestamp() for n in nodes)
+    else:
+        return math.inf
+
+
+def _route_time_partial_cost(nodes, _) -> float:
+    """Partial cost version of above. It is a lower bound to the true cost."""
+    if all(n.has_solution for n in nodes):
+        return max(n.creation_time.timestamp() for n in nodes)
+    else:
+        return math.inf
+
+
+def iter_routes_in_order_found(
```

Comment:
nit: This is very minor but I noticed that `min_cost_routes` and `iter_routes_in_order_found` are similar and both return an iterator over routes, while in a sense they follow different "naming schemes"?

Reply:
Good observation. In hindsight I think `min_cost_routes` should be named `iter_min_cost_routes`, but I didn't want to make breaking changes.

I considered naming the new method `first_routes` instead of `iter_routes_in_order_found`, but I thought that was a bit confusing because it sounds a bit like it returns just the first route.

If you can think of a better name I am happy to rename it, otherwise we can later rename `min_cost_routes` to `iter_min_cost_routes`.

Reply:
> In hindsight I think min_cost_routes should be named iter_min_cost_routes, but I didn't want to make breaking changes.

Yup, that's what I was thinking (or `iter_routes_[something]` but maybe that wouldn't sound natural). Regarding breaking changes, they should be fine given a small number of "users" depending on `syntheseus` downstream (and internal code has `syntheseus` pinned to a particular commit so it won't break by itself), and it's also better sooner than later 😄 But we might find other things worth renaming, so we may want to bundle these together in a PR that purely deals with renaming, and keep this one non-breaking.

Reply:
Let's just rename this now and accept breaking changes. I pushed a change which does this.